cmake/Modules/Sanitizer.cmake

```diff
@@ -0,0 +1,105 @@
+#
+# cmake/Modules/Sanitizers.cmake
+#
+# The MIT License
+#
+# Copyright (c) 2023 TileDB, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+# Sanitizers
+#   https://github.com/google/sanitizers
+#   https://github.com/google/sanitizers/wiki/AddressSanitizer
+#
+# GCC
+#   all: https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html
+#   All the sanitizer options are here, together with all the similar options gcc offers.
+#
+# Clang
+#   address: https://clang.llvm.org/docs/AddressSanitizer.html
+#       Recommends linking with `clang++`. No warning if otherwise at this time.
+#   undefined behavior: https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html
+#   memory: https://clang.llvm.org/docs/MemorySanitizer.html
+#
+# MSVC
+#   Only supports the address sanitizer
+#   https://docs.microsoft.com/en-us/cpp/sanitizers/asan
+#   https://docs.microsoft.com/en-us/cpp/sanitizers/asan-building
+#   https://devblogs.microsoft.com/cppblog/addresssanitizer-asan-for-windows-with-msvc/
+#   https://devblogs.microsoft.com/cppblog/address-sanitizer-for-msvc-now-generally-available/
+
+if(NOT SANITIZER)
+    return()
+endif()
```

Comment:
None of the code in this module runs because `SANITIZER` is never true.

We need a sanitizer option declared in `cmake/Options/BuildOptions.cmake` and the bootstrap script needs to use it.

Reply:
Added build option. I also renamed `SANITIZER` to `TILEDB_SANITIZER` (I guessed we can do it since the CMake variable is undocumented, but I can revert the commit if we can't).